components/playlist/browser/playlist_tab_helper.cc

```diff
@@ -351,6 +399,35 @@ void PlaylistTabHelper::OnFoundMediaFromContents(
   for (auto& observer : observers_) {
     observer.OnFoundItemsChanged(found_items_);
   }
+
+  if (found_items_.size() &&
+      !service_->ShouldExtractMediaFromBackgroundWebContents(found_items_)) {
+    // Wait until we find media or timeout. Some pages might not have media
+    // initially but update media later.
+    auto callbacks =
+        std::move(media_extracted_from_background_web_contents_callbacks_);
+    for (auto& callback : callbacks) {
+      std::move(callback).Run(true);
+    }
+    media_extraction_from_background_web_contents_timer_.Stop();
+  }
+  sent_extract_media_request_ = false;
+}
+
+void PlaylistTabHelper::OnMediaExtractionFromBackgroundWebContentsTimeout() {
+  // Media extraction failed. Found items is blob: that we can't cache from.
+  // Thus, clear them.
+  CHECK(service_->ShouldExtractMediaFromBackgroundWebContents(found_items_));
```

Comment:
what is the purpose of this `CHECK`? It seems very odd to check this here on a callback